src/main/java/StringCalculator.java

```diff
@@ -0,0 +1,83 @@
+/*
+input을 받고 빈문자열인지 null인지 체크
+구분자 체크 -> 구분자 예외 체크
+구분자 대로 문자열 계산기 구현
+ */
+
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    //입력 문자열이 null인지 확인
+    public String checkInput(String input){
+        if(input == null) throw new RuntimeException("null은 허용되지 않습니다.");
+        return input;
+    }
+
+    //커스텀 구분자가 있는지 확인
+    public int checkDelimiter(String checkedInput) {
+        if(checkedInput.isEmpty()) return -2; //문자열이 빈 경우
+
+
+        if (checkedInput.startsWith("//")) { //커스텀 구분자를 설정하는가?
+            int idx = checkedInput.indexOf("\n"); //제대로된 형식인가?
+            if (idx == -1) throw new RuntimeException("잘못된 커스텀 구분자 형식입니다.");
+            return idx; //구분자 위치 전달
+        }
+
+        return -1; //기본 구분자대로 문자열을 나눔
+    }
```

Comment:
커스텀 구분자가 있는지 확인하기 보단, 커스텀 구분자가 를 추출해서 반환해주면 좋을 것 같아요!
`커스텀 구분자가 있으면 커스텀 구분자를 추출하고 아니면 빈 문자열 반환`

Reply:
extractDelimiter과 같은 함수를 만들어서 입력 문자열에서 구분자를 추출하는 함수를 만들어도 좋을 것 같아요.
```suggestion
    public String extractDelimiter(String input) {
        String delimiter = ",:";
        if (checkedInput.startsWith("//")) { //커스텀 구분자를 설정하는가?
            int index = checkedInput.indexOf("\n"); //제대로된 형식인가?
            if (index == -1) throw new RuntimeException("잘못된 커스텀 구분자 형식입니다.");
           delimiter += input.subString(커스텀구분자 시작index, 커스텀구분자 종료index);
        }
        return delimiter;
    }
```

Reply:
checkDelimiter에 맞는 내용만 생각해서 구분자가 있는지 확인하는 함수로만 만들려고 생각한것 같습니다. 알려주셔서 감사합니다!